cns/service/main.go

```diff
@@ -1303,6 +1303,52 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 	// TODO(rbtr): nodename and namespace should be in the cns config
 	directscopedcli := nncctrl.NewScopedClient(directnnccli, types.NamespacedName{Namespace: "kube-system", Name: nodeName})
 
+	// Create the base NNC CRD if HomeAz is enabled
+	if cnsconfig.EnableHomeAz {
+		homeAzResponse := httpRestServiceImplementation.GetHomeAz(ctx)
+		az := homeAzResponse.HomeAzResponse.HomeAz
```

Comment:
This cannot be used as is. When there is a cache miss, the response for GetHomeAz will contain some error values which should be checked. I don't know why the original function doesn't return an error to make this clearer.

Reply:
Updated the client API to handle the cache miss logic and error out and checking for errors in here before updating homeAz value.